src/ethereum/frontier/vm/instructions/memory.py

```diff
@@ -0,0 +1,142 @@
+"""
+Ethereum Virtual Machine (EVM) Memory Instructions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. contents:: Table of Contents
+    :backlinks: none
+    :local:
+
+Introduction
+------------
+
+Implementations of the EVM Memory instructions.
+"""
+
+from ethereum.base_types import U256
+from ethereum.frontier.vm.memory import memory_read_bytes, memory_write
+
+from .. import Evm
+from ..gas import (
+    GAS_BASE,
+    GAS_VERY_LOW,
+    calculate_gas_for_extending_memory,
+    subtract_gas,
+)
+from ..memory import memory_extend
+from ..stack import pop, push
+
+
+def mstore(evm: Evm) -> None:
+    """
+    Stores a word to memory.
+
+    Parameters
+    ----------
+    evm :
+        The current EVM frame.
+
+    Raises
+    ------
+    StackUnderflowError
+        If `len(stack)` is less than `2`.
+    OutOfGasError
+        If `evm.gas_left` is less than
+        `GAS_LOW` + gas needed to extend memeory.
+    """
+    start_position = pop(evm.stack)
+    value = pop(evm.stack).to_be_bytes()
+    padded_value = value.rjust(32, b"\x00")
+    # normalize this to make sure that size doesn't exceed 32 bytes
+    normalized_value = padded_value[-32:]
```

Comment:
I'm still somewhat unsure how this is different from [`U256.to_be_bytes32`][0]. Could you maybe provide an example of how this snippet behaves differently? Should we change the implementation of [`U256.to_be_bytes32`][0]?

[0]: https://github.com/ethereum/eth1.0-specs/blob/3cac857d3556898adfd2ea988adfa6a794a0b367/src/ethereum/base_types.py#L548

Reply:
I was confused earlier because `U256.to_be_bytes32` would raise `OverflowError` for `value > U256_MAX_VALUE`. But now that I think of it that case is not possible as the stack is limited to U256 numbers only.  
Thanks for catching this. I have applied your suggestion.